server/avatars.js

```diff
@@ -1,35 +1,33 @@
 import express from 'express'
 
-import {connect} from 'src/db'
+import {UserAvatar} from 'src/server/services/dataService'
 
 /* eslint new-cap: [2, {"capIsNewExceptions": ["Router"]}] */
 const app = new express.Router()
-const r = connect()
 
-app.get('/:filename', (req, res) => {
+app.get('/:filename', async (req, res) => {
   const {filename} = req.params
   const userId = filename.replace(/\.jpe?g$/, '')
-  r.table('userAvatars')
-    .get(userId)
-    .run()
-    .then(userAvatar => {
-      if (!userAvatar) {
-        const notFoundAvatarURL = 'https://brand.learnersguild.org/assets/echo-icon-128x128.png'
-        res
-          .status(307)
-          .redirect(notFoundAvatarURL)
-        return
-      }
+  let userAvatar
 
-      res
-        .status(200)
-        .set({
-          'Content-Type': 'image/jpeg',
-          'Content-Length': userAvatar.jpegData.length,
-          'Cache-Tag': userAvatar.updatedAt.getTime(),
-        })
-        .send(userAvatar.jpegData)
+  try {
+    userAvatar = await UserAvatar.get(userId)
+  } catch (error) {
+    const notFoundAvatarURL = 'https://brand.learnersguild.org/assets/echo-icon-128x128.png'
+    res
+      .status(307)
+      .redirect(notFoundAvatarURL)
+    return
+  }
+
+  res
+    .status(200)
+    .set({
+      'Content-Type': 'image/jpeg',
+      'Content-Length': userAvatar.jpegData.length,
+      'Cache-Tag': userAvatar.updatedAt.getTime(),
     })
+    .send(userAvatar.jpegData)
 })
```

Comment:
The fetch from the DB could fail for many reasons; we don't want to assume that it will only fail because of a document not found error. I'd recommend doing something like this for now:
```js
import {errors} from 'src/server/services/dataService'

...

const userAvatar = await UserAvatar.get(userId)
  .catch(errors.DocumentNotFound, () => ()) // no-op

if (userAvatar) {
  res
    .status(200)
    .set({
      'Content-Type': 'image/jpeg',
      'Content-Length': userAvatar.jpegData.length,
      'Cache-Tag': userAvatar.updatedAt.getTime(),
    })
} else {
  const notFoundAvatarURL = 'https://brand.learnersguild.org/assets/echo-icon-128x128.png'
  res
    .status(307)
    .redirect(notFoundAvatarURL)
}
```